docs/syntax/imports.md

```diff
@@ -27,40 +27,50 @@ code from modules.
 
 ## Qualified Names
 
-Both imports and exports require the use of qualified module names. A qualified
+In the following text, **entity** shall denote a module, a method (instance,
+static, foreign), type, or a type constructor. In other words, an _entity_ is
+anything that can be assigned to a variable.
+
+Both imports and exports require the use of qualified entity names. A qualified
 name consists of the library namespace (usually organization under which its
 published) and the library name, followed by module names mirroring the source
-tree of the library. For example the file `src/Stuff/Things/Util.enso` inside
-the library `My_Lib` published by the user `wdanilo` would have the following
-qualified name: `wdanilo.My_Lib.Stuff.Things.Util`. To facilitate library
-renaming (or deciding on the publishing organization later in the development
-cycle, or working on a project that won't be published) it is possible to use
-the keyword `project` instead of namespace and project name, to import a file in
-the same project. Therefore, the file `src/Varia/Tools/Manager.enso` in `My_Lib`
-published (or not) by `wdanilo` may use `project.Stuff.Things.Util` to refer to
-the previously mentioned file.
+tree of the library, followed by an entity name within that module. For example
+the file `src/Stuff/Things/Util.enso` inside the library `My_Lib` published by
+the user `wdanilo` would have the following qualified name:
+`wdanilo.My_Lib.Stuff.Things.Util` and the type `My_Type` within that module
+would have the qualified name `wdanilo.My_Lib.Stuff.Things.Util.My_Type`. To
+facilitate library renaming (or deciding on the publishing organization later in
+the development cycle, or working on a project that won't be published) it is
+possible to use the keyword `project` instead of namespace and project name, to
+import a file in the same project. Therefore, the file
+`src/Varia/Tools/Manager.enso` in `My_Lib` published (or not) by `wdanilo` may
+use `project.Stuff.Things.Util` to refer to the previously mentioned file.
+
+Currently, the `project` keyword works only in import and export statements.
+Note that it is possible to export a symbol from the current project with the
+`project` keyword without first importing it.
```

Comment:
Changes in this PR are mentioned just in this paragraph. The rest of the diff to this documentation file is just an attempt to update the specification to correspond to the current functionality.